src/test/java/core/basesyntax/service/operation/BalanceOperationTest.java

```diff
@@ -0,0 +1,99 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.exceptions.RepeatedBalanceOperatorException;
+import core.basesyntax.exceptions.WrongNameException;
+import core.basesyntax.exceptions.WrongQuantityException;
+import core.basesyntax.model.Fruit;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class BalanceOperationTest {
+    private static OperationHandler balanceOperationHandler;
+    private static Fruit fruit;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        balanceOperationHandler = new BalanceOperationHandler();
+        fruit = new Fruit("apple", 14);
+    }
+
+    @Before
+    public void setUp() {
+        FruitStorage.fruits.clear();
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void balance_NullFruit_NotOk() {
+        balanceOperationHandler.doOperation(null);
+    }
+
+    @Test
+    public void balance_FruitWithWrongName_NotOk() {
+        int expectedExceptionNumber = 4;
+        int exceptionNumber = 0;
+        try {
+            balanceOperationHandler.doOperation(new Fruit(null, 14));
+        } catch (NullPointerException e) {
+            exceptionNumber++;
+            try {
+                balanceOperationHandler.doOperation(new Fruit("apple_pie", 14));
+            } catch (WrongNameException ex) {
+                exceptionNumber++;
+                try {
+                    balanceOperationHandler.doOperation(new Fruit("apple0", 14));
+                } catch (WrongNameException exc) {
+                    exceptionNumber++;
+                    try {
+                        balanceOperationHandler.doOperation(new Fruit("-apple", 14));
+                    } catch (WrongNameException exception) {
+                        exceptionNumber++;
+                        try {
+                            balanceOperationHandler.doOperation(new Fruit("apple", 14));
+                        } catch (WrongNameException notException) {
+                            exceptionNumber++;
+                        }
+                    }
+                }
```

Comment:
same lets split this test into several independent tests. (same for other classes)